examples/auction/contract.py

```diff
@@ -82,45 +83,40 @@ def bid(self, pay: PaymentTransaction) -> None:
 
     @arc4.abimethod
     def claim_bids(self) -> None:
-        original_amount = self.claimable_amount[op.Transaction.sender]
-
-        amount = original_amount
+        amount = original_amount = self.claimable_amount[op.Transaction.sender]
 
         # subtract previous bid if sender is previous bidder
         if op.Transaction.sender == self.previous_bidder:
             amount -= self.previous_bid
 
-        op.CreateInnerTransaction.begin()
-        op.CreateInnerTransaction.set_type_enum(TransactionType.Payment)
-        op.CreateInnerTransaction.set_fee(0)  # cover fee with outer txn
-        op.CreateInnerTransaction.set_receiver(op.Transaction.sender)
-        op.CreateInnerTransaction.set_asset_amount(amount)
-        op.CreateInnerTransaction.submit()
+        itxn.PaymentTransactionParams(
+            fee=0,
+            amount=amount,
+            receiver=op.Transaction.sender,
+        ).submit()
 
         self.claimable_amount[op.Transaction.sender] = original_amount - amount
 
     @arc4.abimethod
     def claim_asset(self, asset: Asset) -> None:
         assert op.Global.latest_timestamp > self.auction_end, "auction has not ended"
-
+        previous_bidder = self.previous_bidder
```

Comment:
Can we do this in the compiler rather than expecting the user to know it needs to be copied to a local? 🤔

Reply:
or is the copy to local just an optimization to prevent global state reads? is that something we need to do?

Reply:
Yeah this change isn't required for the inner transaction to work, was just a minor optimisation to prevent two reads (and associated asserts). But I'll remove the changes from this PR, as it does make it seem like that.

There is probably an optimisation here that could be done to treat duplicate state reads as a repeated sub expression if there is no write between them